pkg/utils/kubernetes/self_metadata.go

```diff
@@ -0,0 +1,61 @@
+package kubernetes
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// This file includes utility functions for operating the metadata of
+// the running KO instance itself.
+
+const (
+	podNamespaceEnvName         = "POD_NAMESPACE"
+	podLabelsFile               = "/etc/podinfo/labels"
+	serviceAccountNamespaceFile = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+)
+
+// GetSelfNamespace gets the namespace in which KO runs.
+func GetSelfNamespace() (string, error) {
+	if ns := os.Getenv(podNamespaceEnvName); ns != "" {
+		return ns, nil
+	}
+	// This actually gets the namespace of the service account to run the pod.
+	buf, err := os.ReadFile(serviceAccountNamespaceFile)
+	if err != nil {
+		return "", err
+	}
+	return string(buf), nil
+}
+
+// GetSelfPodLabels gets all the labels of the KO pod.
+func GetSelfPodLabels() (map[string]string, error) {
+	buf, err := os.ReadFile(podLabelsFile)
+	if err != nil {
+		return nil, fmt.Errorf("cannot find pod labels from file %s: %w", podLabelsFile, err)
+	}
+
+	labelList := strings.Split(string(buf), "\n")
+	ret := make(map[string]string, len(labelList))
+	for _, label := range labelList {
```

Comment:
We could split using the iterator (that would prevent us being able to use the map size hint but that's rarely useful, and doesn't impact performance much, especially on small sized maps like here).

```suggestion
	labels := strings.SplitSeq(string(buf), "\n")
	ret := make(map[string]string)
	for label := range labels {
```